moonraker/components/file_manager/metadata.py

```diff
@@ -747,33 +747,65 @@ def parse_nozzle_diameter(self) -> Optional[float]:
 
 class IceSL(BaseSlicer):
     def check_identity(self, data) -> Optional[Dict[str, Any]]:
-        match = re.search(r"; <IceSL.*>", data)
+        match = re.search(r"<IceSL\s(.*)>", data)
```

Comment:
By default, in the settings dump that is produced, it is indeed a build date.
However, since we can easily manage what a profile will output in the gcode, we can extend the header by providing additional information.

As implemented in the [Voron_V0](https://github.com/shapeforge/icesl-printers/blob/master/fff/Voron_V0/printer.lua) profile (which will set a "standard" to extend a profile for Klipper), following our [documentation](https://gitlab.inria.fr/mfx/icesl-documentation/-/wikis/Printer-profile#state-variables), we can output additional information (see the `header()` function, from line 44 to 57, where at line 48 we can get the version from *utility state variable*).

In addition to that, I added the condition to fill the `version` variable to provide compatibility with *non-extended-profiles*, which will ignore the date if there is no version.